File: clog-dart/lib/db/repo/src/service_log_repo.dart
/*
 * Copyright (c) 2022.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

part of db_repo;

class _ServiceLogRepo {
  Future<FError> insert(ServiceLogEntity e) async {
    final sql = '''
INSERT INTO service_log
  (id, uid, user_id, partner_id, xid,
   svc_name, svc_version, svc_parent, endpoint, version,
   message, severity, path, function, req_header,
   req_body, req_par, res_data, res_code, data,
   error, stack_trace, client_ip, duration_ms, start_at,
   finish_at, created_at)
VALUES
  (@id, @uid, @user_id, @partner_id, @xid,
   @svc_name, @svc_version, @svc_parent, @endpoint, @version,
   @message, @severity, @path, @function, @req_header,
   @req_body, @req_par, @res_data, @res_code, @data,
   @error, @stack_trace, @client_ip, @duration_ms, @start_at,
   @finish_at, @created_at)
''';

    final pars = {
      'id': e.id,
      'uid': e.uid,
      'user_id': e.userId,
      'partner_id': e.partnerId,
      'xid': e.xid,
      'svc_name': e.svcName,
      'svc_version': e.svcVersion,
      'svc_parent': e.svcParent,
      'endpoint': e.endpoint,
      'version': e.version,
      'message': e.message,
      'severity': e.severity,
      'path': e.path,
      'function': e.function,
      'req_header': e.reqHeader,
      'req_body': e.reqBody,
      'req_par': e.reqPar,
      'res_data': e.resData,
      'res_code': e.resCode,
      'data': e.data,
      'error': e.error,
      'stack_trace': e.stackTrace,
      'client_ip': e.clientIP,
      'duration_ms': e.durationMs,
      'start_at': e.startAt,
      'finish_at': e.finishAt,
      'created_at': e.createdAt,
    };

    return await app.db.execute(sql: sql, pars: pars);
  }
}

File: clog-go/db/repo/service-log-repo.go
/*
 * Copyright (c) 2023.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

package repo

import (
	"clog/app"
	"clog/db/entity"

	"github.com/andypangaribuan/project9/core/db"
)

type srServiceLog struct {
	repo *db.Repo[entity.ServiceLog]
}

func (slf srServiceLog) new() *srServiceLog {
	slf.repo = new(db.Repo[entity.ServiceLog])
	slf.repo.DbInstance = app.Db
	slf.repo.PrintUnsafeErr = !app.Env.AppEnv.IsProd()

	slf.repo.TableName = app.Env.TableName.ServiceLog
	slf.repo.SetInsertColumnNames(`
		id, uid, user_id, partner_id, xid,
		svc_name, svc_version, svc_parent, endpoint, version,
		message, severity, path, function, req_header,
		req_body, req_par, res_data, res_code, data,
		error, stack_trace, client_ip, duration_ms, start_at,
		finish_at, created_at`)
	slf.repo.InsertParamSign = slf.repo.GenerateParamSigns(slf.repo.InsertColumnNames)

	return &slf
}

func (slf *srServiceLog) Insert(e entity.ServiceLog) error {
	return slf.repo.Insert(nil,
		e.Id, e.Uid, e.UserId, e.PartnerId, e.Xid,
		e.SvcName, e.SvcVersion, e.SvcParent, e.Endpoint, e.Version,
		e.Message, e.Severity, e.Path, e.Function, e.ReqHeader,
		e.ReqBody, e.ReqPar, e.ResData, e.ResCode, e.Data,
		e.Error, e.StackTrace, e.ClientIp, e.DurationMs, e.StartAt,
		e.FinishAt, e.CreatedAt,
	)
}
